### PR TITLE
Add support for YMD75 rev 2

### PR DESCRIPTION
**Pull request by jaredh**
## Description

Adds support for revision 2 of the YMD75 keyboard from YMDK.

This PR refactors the original `ymd75` keyboard support into `rev1` and introduces `rev2` which YMKD has been selling recently.

Unfortunately I don't have a `rev1`, so I cannot test my changes on that board. I do however, have two `rev2` PCBs. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 🙅‍♂️

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jaredh** [comment]
Don't have permissions I guess to add reviewers but pinging keyboard maintainer @WarmCatUK and @zvecr who pointed me in the right direction.

File: keyboards/ymd75/rev1/rules.mk [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+CONSOLE_ENABLE = no
+COMMAND_ENABLE = yes
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = i2c
+NKRO_ENABLE = no
+# Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+
+DISABLE_WS2812 = no
+
```
Comment (zvecr) [inline keyboards/ymd75/rev1/rules.mk:27]:
```suggestion

```

File: keyboards/ymd75/rev2/rev2.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#ifndef KEYMAP_COMMON_H
+#define KEYMAP_COMMON_H
+
+#include "quantum.h"
+#include "quantum_keycodes.h"
+#include "keycode.h"
+#include "action.h"
```
Comment (zvecr) [inline keyboards/ymd75/rev2/rev2.h:7]:
```suggestion
#pragma once

#include "quantum.h"
```

File: keyboards/ymd75/rev2/rev2.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+) { \
+	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014 }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  KC_NO, K114 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  K213,  K214 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  KC_NO, K313,  K314 }, \
+	{ K400,  KC_NO, K402,  K403,  K404,  K405,  K406,  K407,  K408,  K409,  K410,  K411,  K412,  K413,  K414 }, \
+	{ K500,  K501,  K502,  KC_NO, KC_NO, K505,  K506,  K507,  K508,  K509,  K510,  K511,  KC_NO, K513,  K514 }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/ymd75/rev2/rev2.h:25]:
```suggestion
```

File: keyboards/ymd75/rev2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+CONSOLE_ENABLE = no
+COMMAND_ENABLE = yes
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = i2c
+NKRO_ENABLE = no
+# Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+
+DISABLE_WS2812 = no
+
```
Comment (zvecr) [inline keyboards/ymd75/rev2/rules.mk:27]:
```suggestion

```

File: keyboards/ymd75/rev2/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+// GBP £ Macro (sends alt 156 - windows users only)
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch(keycode) {
+            case P_MACRO:
+                SEND_STRING(SS_DOWN(X_LALT));
+                SEND_STRING(SS_TAP(X_KP_1));
+                SEND_STRING(SS_TAP(X_KP_5));
+                SEND_STRING(SS_TAP(X_KP_6));
+                SEND_STRING(SS_UP(X_LALT));
```
Comment (fauxpark) [inline keyboards/ymd75/rev2/keymaps/default/keymap.c:89]:
```suggestion
                register_code(KC_LALT);
                tap_code(KC_KP_1);
                tap_code(KC_KP_5);
                tap_code(KC_KP_6));
                unregister_code(KC_LALT);
```

File: keyboards/ymd75/rev2/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                return false; break;
```
Comment (fauxpark) [inline keyboards/ymd75/rev2/keymaps/default/keymap.c:90]:
```suggestion
                return false;
```

**Comment by jaredh** [comment]
I had a fundamental misunderstanding of the `LAYOUT` macros which made me think I had to provide a separate keymap for each revision. I've addressed the `rev2` `LAYOUT` issue and now use a single default keymap.

Thanks for the quick feedback and suggestions. Well documented project too!

File: keyboards/ymd75/rev2/rules.mk [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = bootloadHID
```
Comment (fauxpark) [inline keyboards/ymd75/rev2/rules.mk:12]: This stuff can stay in the top-level rules.mk (at least until YMDK make a rev3 or something...)

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**